new_client/src/components/events/AddEvent.jsx

```diff
@@ -0,0 +1,250 @@
+import React, { Component } from 'react';
+import axios from 'axios';
+
+import '../auth/Error.css';
+
+class AddEvent extends Component {
+  state = {
+    event: {
+      eventName: '',
+      description: '',
+      date: '',
+      venue: '',
+      graduationYear: '',
+      selectedFile: null
+    },
+    error: {
+      eventNameError: '',
+      descriptionError: '',
+      dateError: '',
+      venueError: '',
+      graduationYearError: ''
+    }
+  };
+
+  onFileChange = (e) => {
+    const event = { ...this.state.event };
+    event.selectedFile = e.target.files[0];
+    this.setState({ event: event });
+  };
+
+  handleChange = (e) => {
+    const field = e.target.name;
+
+    const event = { ...this.state.event, [field]: e.target.value };
+
+    this.setState({
+      event: event
+    });
+  };
+
+  isValid = () => {
+    const error = {
+      ...this.state.error
+    };
+    let ret = true;
+    if (this.state.event.eventName === '') {
+      error.eventNameError = '*Event name cannot be empty';
+      ret = false;
+    }
+    if (this.state.event.description === '') {
+      error.descriptionError = '*Event description cannot be empty';
+      ret = false;
+    }
+    if (this.state.event.venue === '') {
+      error.venueError = '*Event venue cannot be empty';
+      ret = false;
+    }
+    if (this.state.event.date === '') {
+      error.dateError = '*Event date cannot be empty';
+      ret = false;
+    }
+    if (this.state.event.graduationYear === '') {

```

Comment:
Since graduationYear is a number, it would be better if this was to be compared with null, rather than an empty string.